neoemf-core/src/main/java/fr/inria/atlanmod/neoemf/core/internal/LazyStoreList.java

```diff
@@ -0,0 +1,416 @@
+        }
+        else {
+            delegateUnset();
+        }
+    }
+
+    protected boolean delegateIsSet() {
+        return eStore().isSet(owner, eStructuralFeature);
+    }
+
+    protected void delegateUnset() {
+        eStore().unset(owner, eStructuralFeature);
+    }
+
+    @Override
+    protected List<E> delegateList() {
+        throw new UnsupportedOperationException("delegateList");
+    }
+
+    @Override
+    protected int delegateSize() {
+        return eStore().size(owner, eStructuralFeature);
+    }
+
+    @Override
+    protected boolean delegateIsEmpty() {
+        return eStore().isEmpty(owner, eStructuralFeature);
+    }
+
+    @Override
+    protected boolean delegateContains(Object object) {
+        return eStore().contains(owner, eStructuralFeature, object);
+    }
+
+    @Override
+    protected boolean delegateContainsAll(Collection<?> collection) {
+        if (isNull(collection) || collection.isEmpty()) {
+            return false;
+        }
+
+        if (collection.size() == 1) {
+            return contains(collection.iterator().next());
+        }
+
+        return eStore().getAll(owner, eStructuralFeature).containsAll(collection);
+    }
+
+    @Override
+    protected int delegateIndexOf(Object object) {
+        return eStore().indexOf(owner, eStructuralFeature, object);
+    }
+
+    @Override
+    protected int delegateLastIndexOf(Object object) {
+        return eStore().lastIndexOf(owner, eStructuralFeature, object);
+    }
+
+    @Override
+    protected Object[] delegateToArray() {
+        return eStore().toArray(owner, eStructuralFeature);
+    }
+
+    @Override
+    protected <T> T[] delegateToArray(T[] array) {
+        return eStore().toArray(owner, eStructuralFeature, array);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected E delegateGet(int index) {
+        return (E) eStore().get(owner, eStructuralFeature, index);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected E delegateSet(int index, E object) {
+        return (E) eStore().set(owner, eStructuralFeature, index, object);
+    }
+
+    @Override
+    protected void delegateAdd(Object object) {
+        delegateAdd(InternalEObject.EStore.NO_INDEX, object);
+    }
+
+    @Override
+    protected void delegateAdd(int index, Object object) {
+        eStore().add(owner, eStructuralFeature, index, object);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected E delegateRemove(int index) {
+        return (E) eStore().remove(owner, eStructuralFeature, index);
+    }
+
+    @Override
+    protected void delegateClear() {
+        eStore().clear(owner, eStructuralFeature);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected E delegateMove(int targetIndex, int sourceIndex) {
+        return (E) eStore().move(owner, eStructuralFeature, targetIndex, sourceIndex);
+    }
+
+    @Override
+    // FIXME Does not resolve
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1134)